scico/loss.py

```diff
@@ -218,40 +228,43 @@ def __init__(
         if isinstance(self.A, linop.LinearOperator):
             self.is_quadratic = True
 
-        if isinstance(self.A, linop.Diagonal) and isinstance(self.weight_op, linop.Diagonal):
+        if isinstance(self.A, linop.Diagonal) and isinstance(self.W, linop.Diagonal):
             self.has_prox = True
 
     def __call__(self, x: Union[JaxArray, BlockArray]) -> float:
-        return self.scale * self.functional(self.weight_op(self.y - self.A(x)))
+        return self.scale * (self.W.diagonal * snp.abs(self.y - self.A(x)) ** 2).sum()
 
```

Comment:
See earlier comment on similar lines.

Reply:
> CI seems to be failing because the changes would slightly reduce the test coverage percentage.

Let's see whether that is still the case when we add the test for the Hessian.

Reply:
Still unhappy, it seems. It would be best to address this before we merge.

Reply:
I added a test for `loss.PoissonLoss` which will, I assume, resolve this.

@Michael-T-McCann : Would you not agree that the `Loss` tests should be in a separate `test_loss.py` file rather than included in `test_functional.py`?

Reply:
`Loss` is currently a subclass of `Functional` (was not always this way). Therefore I think it makes sense for the losses to get tested in `test_functional.py`, unless the file is much too long.